### DISCUSSION TITLE
Can I combine SOFA with a robot in Gazebo and ROS?

### DISCUSSION THREAD

**Discussion by Heidari60** [sofa-framework/sofa#3916] [Ideas / Suggestions]
I hope you are fine

Can I use a continuum manipulator and combine it with a robot in Gazebo and ROS?

Do you have any examples of that?

Sincerely,
Hamidreza

**Comment by damienmarchal** [comment]
Hi @Heidari60,

To connect Sofa with a ROS2 ecosystem you can use the example code that is provided in the SoftRobots plugin: 
https://github.com/SofaDefrost/SoftRobots/tree/master/docs/sofapython3/examples/sofaros

It is technically possible, despite far from trivial, to use sofa to implement the simulation backend of Gazebo, but currently I didn't hear about someone having done such thing.  Currently at DEFROST we found easier to implement our complete simulation in Sofa and are investigating to add support for the Pinochio library: https://github.com/stack-of-tasks/pinocchio to improve how we handle poly-articulated rigid bodies.
